### ISSUE TITLE
Dockerize the application

### ISSUE THREAD

**Issue by tiagostutz** [bancodobrasil/stop-analyzing-embed#19]
To have a deployment ready application, it would be great to have it Dockerized. To do so, we should have, basically 5 things:
1. **Create a Dockerfile**
1. **A `docker-compose.yml` with `build` directive to run locally**
1. **README with instructions on how to ruin and build locally**
1. A configured CI that will be triggered to build the image in DockerHub as new releases are created here in the repo
1. README with a `docker-compose.yml` instructions (including the use of stop-analyzing-api)

For this issue, we are building the first 3 items. Is that ok?

**Comment by giteshk11** [comment]
@tiagostutz  can I try this?

**Comment by tiagostutz** [comment]
Sure, @giteshk11 !
Just assigned to you. If you need any help, let me know.

Just a suggestion, there's this light web server that you could consider: https://github.com/lwsjs/lws/, but feel free to bring your preferences.

**Comment by giteshk11** [comment]
@tiagostutz just a quick question, do we need any OS as a base in the container?

**Comment by tiagostutz** [comment]
In most cases we need an OS, prefering the smaller ones like alpine or debian stretch-slim, but if you have any option that brings up a web server from scratch, that's ok also. Also, you could search for a ready-to-go Docker imagem with a web server.

What web server are you planning to use?

**Comment by giteshk11** [comment]
I'm thinking of Nginx or the link you shared before we can use that too.

**Comment by tiagostutz** [comment]
Well, I think we can use this opportunity to increment the open source experience a little bit. What about opening a issue on https://github.com/lwsjs/local-web-server asking whether there is an official Docker image and offering to build it if there isn't one? What do you think?

**Comment by giteshk11** [comment]
Sounds good to me.

**Comment by giteshk11** [comment]
Hi @tiagostutz, I'm having some issues with docker on my machine and I'm not able to figure out. Can you assign this to someone else?

**Comment by tiagostutz** [comment]
Sure! I'll create another issue and assign it to you.

**Comment by lucasravy** [comment]
Hey guys, I would like to try. Can I work on this?
:grin:

**Comment by tiagostutz** [comment]
You got this, @lucasravy .!

Wanna try to open the issue in the local-web-server project?

**Comment by lucasravy** [comment]
@tiagostutz nice! Thanks for the challenge.

I'm working on it right now. As soon as I get the issue's number from [local-web-server](https://github.com/lwsjs/local-web-server) repository, I'll let you know here. 

:nerd_face: :computer:

**Comment by lucasravy** [comment]
The [issue](https://github.com/lwsjs/local-web-server/issues/151) has just been opened!

Quick question: If the official image docker proposal don't work, we're gonna create our own using the same solution, right? So: **alpine** as S.O base + **local-web-server** as web server. Is it good for a start?

**Comment by tiagostutz** [comment]
> The [issue](https://github.com/lwsjs/local-web-server/issues/151) has just been opened!
> 
> Quick question: If the official image docker proposal don't work, we're gonna create our own using the same solution, right? So: **alpine** as S.O base + **local-web-server** as web server. Is it good for a start?

If it doesn't work, I suggest you create the image on your own user and then use it to Dockerize this application here. Sounds good?

**Comment by lucasravy** [comment]
> > The [issue](https://github.com/lwsjs/local-web-server/issues/151) has just been opened!
> > 
> > Quick question: If the official image docker proposal don't work, we're gonna create our own using the same solution, right? So: **alpine** as S.O base + **local-web-server** as web server. Is it good for a start?
> 
> If it doesn't work, I suggest you to create the image on your own user and then use it to Dockerize this application here. Sounds good?

Sounds great! Deal!

**Comment by tiagostutz** [comment]
Hi @lucasravy !

I think we waited enough time for the reply of **local-web-server** maintainers, right? 

What do you think of opening a repo on your own GitHub user for this Docker image with a great README, publishig it to https://hub.docker.com and referencing it in the issue you opened?

**Tip for the README**: https://codarium.substack.com/p/my-secret-sauce-to-make-github-repositories

**Comment by lucasravy** [comment]
Hey @tiagostutz!

I've already done [that](https://hub.docker.com/repository/docker/lucasravy/local-web-server-alpine) (except the great README tips, thanks btw :grinning:) and I'm testing the image on our project. I'm having some troubles with the image size but I think I have the solution and I'll be able to open a PR in the next couple of days.

**Comment by tiagostutz** [comment]
Fantastic, @lucasravy !

Can't wait to check it out. 🤓
